### ISSUE TITLE
StackOverflowErrors have truncated stack traces

### ISSUE THREAD

**Issue by nex3** [dart-lang/sdk#23614]
When the VM detects a `StackOverflowError`, it generates an error whose last line is truncated. To reproduce:

```dart
void recursive() => recursive();

void main() {
  try {
    recursive();
  } catch (error, stackTrace) {
    print(stackTrace);
  }
}
```

The resulting stack trace:

```
#0      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart)
#1      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#2      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#3      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#4      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#5      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#6      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#7      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#8      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#9      recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#10     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#11     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#12     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#13     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
...
...
#14     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#15     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#16     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#17     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#18     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#19     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#20     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#21     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#22     recursive (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:1:21)
#23     main (file:///usr/local/google/home/nweiz/goog/pkg/stack_trace/test.dart:5:5)
#24     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:261)
#25     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.da
```

Note how the last line cuts off in the middle of the filename.

**Comment by nex3** [comment]
Interestingly, as far as I can tell, it seems like it *always* cuts off the last line in the same position.

**Comment by sgmitrovic** [comment]
Fixed in d9860cb6bce5a82da312fc8399a0a49d9b0a813f

**Comment by nex3** [comment]
Thanks!
